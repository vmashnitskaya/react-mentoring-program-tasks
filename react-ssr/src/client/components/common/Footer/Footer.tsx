import React, { FunctionComponent } from 'react';
import loadable from '@loadable/component';
import './Footer.scss';

const Logo = loadable(() => import('../Logo/Logo'));

const Footer: FunctionComponent = () => {
    return (
        <footer>
            <Logo />
        </footer>
    );
};

export default Footer;
