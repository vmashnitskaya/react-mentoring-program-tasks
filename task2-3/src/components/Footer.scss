@import "../constants";

footer {
  position: relative;
  width: 100%;
  height: 60px;
  background-color: $main-grey-light;

  .logo {
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
  }
}
