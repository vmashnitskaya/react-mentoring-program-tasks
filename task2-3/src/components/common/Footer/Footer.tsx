import React, { FunctionComponent } from 'react';
import Logo from '../Logo/Logo';
import './Footer.scss';

const Footer: FunctionComponent = () => {
    return (
        <footer>
            <Logo />
        </footer>
    );
};

export default Footer;
