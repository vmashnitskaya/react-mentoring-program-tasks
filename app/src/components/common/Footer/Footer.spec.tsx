import React from 'react';
import { render } from '@testing-library/react';
import Footer from "./Footer";

describe('Footer', () =>{
    test('renders footer snapshot', () => {
        const {asFragment} = render(<Footer />);

        expect(asFragment()).toMatchSnapshot();
    });
});
