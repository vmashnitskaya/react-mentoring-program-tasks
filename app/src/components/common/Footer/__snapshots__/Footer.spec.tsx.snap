// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footer snapshot 1`] = `
<DocumentFragment>
  <footer>
    <a
      class="logo"
      href="/home"
    >
      <span
        class="logo-netflix"
      >
        netflix
      </span>
      <span>
        roulette
      </span>
    </a>
  </footer>
</DocumentFragment>
`;
